### PR TITLE
helm: Add option to ignore checksumming config/secrets

### PR DESCRIPTION
**Pull request by funkypenguin**
## Description

This PR adds a value `ignoreChartChecksums`, which causes the deployment/statefulset **not** to add the checksum of the config/secret as annotations to the pod template. 

The reason for the checksums is presumably so that when a user changes a config/secret-related value, and applies a `helm upgrade`, the deployment/statefulset are changed, and the pods are updated. 

## Motivation and Context

In my use case, I use this chart as a sub-chart, and I deploy the umbrella chart using Flux v2. Presumably due to the way flux itself adds labels/annotations to deployed objects, every time flux deploys an update to my umbrella chart, even if nothing in the minio sub-chart's values have changed, the checksums of the secret/config.yaml template files change, and my minio pods restart.

This is disruptive and unnecessary, given no changes actually impacted Minio.

## How to test this PR?

I tested the change in a local KinD cluster, and confirmed that setting `ignoreChartChecksums` to `true` allows the deployment/statefulset to be created as expected.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [X] Documentation updated (README just points the reader to `values.yaml`, which is the only place I've documented this. Happy to add details as necessary)
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: helm/minio/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: Multi-Cloud Object Storage
 name: minio
-version: 3.5.7
+version: 3.5.8
```
Comment (harshavardhana) [inline helm/minio/Chart.yaml]: you do not need to change versions @funkypenguin

**Review by harshavardhana — CHANGES_REQUESTED** [review]
rest of the changes lgtm

File: helm/minio/Chart.yaml (hunk 1 above)
Comment (funkypenguin) [reply helm/minio/Chart.yaml]: OK, updated to master's current version (3.5.9), hope that was the correct resolution!

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**